cornell-dti/course-plan
# Prod: Compound Requirements, Graduate Program Support, Cypress, DeleteCourseModal, Setup for AP/IB Overridden Requirements, Typing Dropdowns

### Summary

This pull request includes new features: compound requirements, graduate program support, typing dropdowns. It also includes graph computation setup for AP/IB Overridden Requirements and renaming the reset course modal to be the DeleteCourseModal. Cypress frontend unit testing would be integrated into the CI as of this PR.

- [x] Compound Requirements (EN liberal arts and Math minor migrated to use compound reqs)
![image](https://user-images.githubusercontent.com/54298311/119571626-90a28b80-bd7f-11eb-993b-d057a90ec212.png)

- [x] Grad program support
- [x] First grad program added (MPA) 
![image](https://user-images.githubusercontent.com/54298311/119571688-a7e17900-bd7f-11eb-9746-f8873eea8237.png)

- [x] Graph computation setup for AP/IB Overridden Requirements 
- [x] Cypress frontend unit testing
- [x] Typing Dropdowns on first page of Onboarding
![image](https://user-images.githubusercontent.com/54298311/119571743-bf206680-bd7f-11eb-82ef-e3d582752dd1.png)
- [x] Rename reset course modal to DeleteCourseModal
![image](https://user-images.githubusercontent.com/54298311/119571933-fa229a00-bd7f-11eb-9dfa-f0864ad47e9d.png)


Remaining TODOs:
- [ ] Flesh out frontend for AP/IB Overriding. #514 removes the button to open the slot settings menu for this push to prod, but the button should be reinstated when finishing the (AP/IB) Requirement Overriding feature (#512).


### Test Plan 
1. Switch to EN and make sure you can fulfill the Liberal Arts requirement correctly.
2. Remove the gradPrograms field from your document in the user-onboarding-data collection on Firestore (to simulate an existing user) and make sure it doesn’t break. 
3. Try adding the MPA grad program and make sure you can complete it.
4. Try typing the college/major/minor/grad program to see the dropdown choices filter.
5. Delete a course on the requirements bar by clicking the “Delete >” button.
